src/opm/output/eclipse/EclipseIO.cpp

```diff
@@ -113,14 +111,16 @@ void EclipseIO::Impl::writeINITFile(const data::Solution&                   simP
 
 
 void EclipseIO::Impl::writeEGRIDFile( const NNC& nnc ) {
-    const auto& ioConfig = this->es.getIOConfig();
+    const auto formatted = this->es.cfg().io().getFMTOUT();
 
-    std::string  egridFile( ERT::EclFilename( this->outputDir,
-                                              this->baseName,
-                                              ECL_EGRID_FILE,
-                                              ioConfig.getFMTOUT() ));
+    const auto ext = std::string{ '.' }
+        + (formatted ? std::string{ "F" } : std::string{})
+        + "EGRID";
 
-    this->grid.save( egridFile, ioConfig.getFMTOUT(), nnc, this->es.getDeckUnitSystem());
+    const auto egridFile = (boost::filesystem::path{ this->outputDir }
+        / (this->baseName + ".REPLACE")).replace_extension(ext);
```

Comment:
I guess there is a good reason - but this `".REPLACE"` construction looked funny.

Reply:
It handles the (very) special case of a simulation run named `CASE.123.DATA` for which we've already split off the `.DATA` part.

Reply:
I still understand - but I guess it is fine.

Reply:
> I still don't understand - but I guess it is fine.

I was about to go into a lengthy explanation of the original reasoning, but then I realized that this ultimately doesn't actually do anything other than
```C++
this->baseName + ext
```
so I'll switch to the simpler alternative.